R/tag_query.R

```diff
@@ -597,280 +501,244 @@ tagQuery_ <- function(
-        #' elements.
+        #' * `$parents(cssSelector = NULL)`: Get the
+        #' ancestors of each selected tag, optionally filtered by a
+        #' `cssSelector`.
         parents = function(cssSelector = NULL) {
           rebuild_()
           newTagQuery(
             tagQueryFindParents(selected_, cssSelector)
           )
         },
-        #' * `$closest(cssSelector = NULL)`: For each selected element, get the
-        #' closest ancestor element (including itself) that matches the
-        #' single-element CSS selector. If `cssSelector = NULL`, it is
-        #' equivalent to calling `$selected()`. A new `tagQuery()` object will be
-        #' created with the selected items set to the closest matching elements.
-        closest = function(cssSelector = NULL) {
+        #' * `$closest(cssSelector)`: For each selected tag,
+        #' get the closest ancestor tag (including itself) satisfying a
+        #' `cssSelector`.
+        closest = function(cssSelector) {
           rebuild_()
           newTagQuery(
             tagQueryFindClosest(selected_, cssSelector)
           )
         },
-        #' * `siblings(cssSelector = NULL)`: Get the siblings of each element in
-        #' the set of matched elements. If a CSS selector is provided, only the
-        #' siblings matching the single-element CSS selector will be selected. A
-        #' new `tagQuery()` object will be created with the selected items set
-        #' to the sibling elements.
-        siblings = function(cssSelector = NULL) {
-          rebuild_()
-          newTagQuery(
-            tagQueryFindSiblings(selected_, cssSelector)
-          )
-        },
-        #' * `$filter(fn)`: Update the selected elements to a subset of the
-        #' selected elements given `fn(el, i)` returns `TRUE`. If `fn` is a CSS
-        #' selector, then the selected elements will be filtered if they match
-        #' the single-element CSS selector. A new `tagQuery()` object will be
-        #' created with the selected items set to the filtered selected
-        #' elements.
+        #' ### Custom filter
+        #'
+        #' * `$filter(fn)`: Filter the selected tags to those for which `fn(x,
+        #' i)` returns `TRUE`. In addition to an R function with two arguments
+        #' (the selected tag `x` and the index `i`), `fn` may also be a valid
+        #' CSS selector.
         filter = function(fn) {
           rebuild_()
           newTagQuery(
             tagQueryFindFilter(selected_, fn),
             rebuild = TRUE
           )
         },
-        #' * `$reset()`: A new `tagQuery()` object will be created with the
-        #' selected items set to the top level tag objects.
-        reset = function() {
+        #' ### Reset
+        #'
+        #' * `$resetSelection()`: Reset selected tags to the `$root()` tag. Useful
+        #' in combination with `$replaceWith()` since it empties the selection.
+        resetSelection = function() {
```

Comment:
Undo this change and rebase off of #235 when merged.

Also, update vignette to use this new name